lib/html5ever/precompiled.ex

```diff
@@ -265,6 +335,56 @@ defmodule Html5ever.Precompiled do
     end
   end
 
+  defp checksum_map(module_name) when is_atom(module_name) do
+    module_name
+    |> checksum_file()
+    |> read_map_from_file()
+  end
+
+  defp check_file_integrity(file_path, module_name) when is_atom(module_name) do
+    module_name
+    |> checksum_map()
+    |> check_integrity_from_map(file_path, module_name)
+  end
+
+  # It receives the map of %{ "filename" => "algo:checksum" } with the file path
+  def check_integrity_from_map(checksum_map, file_path, module_name) do
+    basename = Path.basename(file_path)
+
+    case Map.fetch(checksum_map, basename) do
+      {:ok, algo_with_hash} ->
+        [algo, hash] = String.split(algo_with_hash, ":")
+        algo = String.to_existing_atom(algo)
+
+        if algo in @checksum_algorithms do
+          with {:ok, content} <- File.read(file_path) do
+            tar_gz_hash =
+              algo
+              |> :crypto.hash(content)
+              |> Base.encode16(case: :lower)
+
+            if hash == tar_gz_hash do
+              :ok
+            else
+              {:error, "the integrity check failed because the checksum of files does not match"}
+            end
+          end
+        else
+          {:error,
+           "checksum algorithm is not supported: #{inspect(algo)}. The supported ones are:\n - #{Enum.join(@checksum_algorithms, "\n - ")}"}
+        end
+
+      :error ->
+        {:error,
+         "the precompiled NIF file does not exist in the checksum file. Please consider run: `mix rustler.download #{inspect(module_name)} --only-local` to generate the checksum file."}
```

Comment:
When would this error happen? Are we sure we want to tell users to run this command?

Reply:
This can happen if the user is trying another target, but he/she didn't generate the checksum for that target. This is going to happen only in development (in the git repo).